Savanna - Iteration 1

EVIJA-OZO opened the pull request:
Pressing buttons A and L adds animals to the field.
Two animal types: Antelope and Lion.
Animals freely move on the field and try to:
a. Antelope to avoid lion,
b. Lion to catch Antelope.
Each animal has vision range and moves based on it.

EVIJA-OZO committed:
Pressing buttons A and L adds animals to the field
Two animal types: Antelope and Lion
Animals freely move on the field and try to:
a. Antelope to avoid lion
b. Lion to catch Antelope
Each animal has vision range and moves based on it

GuntarsD commented on Savanna/View/UserInterface.cs:
```diff
@@ -0,0 +1,38 @@
+        /// </summary>
+        /// <param name="args">Array containing the game field.</param>
+        /// <param name="game">Current game generation.</param>
+        public static void Display(string[,] args, Game game)
```
Generic name. Here and in other places replace with more descriptive one. May be:
gameField

EVIJA-OZO replied:
Renamed.

GuntarsD commented on Savanna/Model/Game.cs:
```diff
@@ -0,0 +1,76 @@
+        /// <summary>
+        /// Checks if cell is on the board.
+        /// </summary>
+        /// <param name="newXCoordinate">New X coordinate.</param>
```
It is better not use generic x and y. Better to use row and column.

EVIJA-OZO replied:
Renamed.

EdgarsBondars commented on Savanna/Model/Animal.cs:
```diff
@@ -0,0 +1,196 @@
+                {
+                    animals.Add(newAnimal);
+                }
+                else
```
else block can be removed.

EVIJA-OZO replied:
Code part removed.

GuntarsD commented on Savanna/Model/Animal.cs:
```diff
@@ -0,0 +1,196 @@
+        public static void AddAnimal(Animal newAnimal, List<Animal> animals)
+        {
+            Random random = new();
+
```
Here and in other places remove obsolete empty lines

EVIJA-OZO replied:
Empty lines checked and removed.

EdgarsBondars commented on Savanna/Model/Animal.cs:
```diff
@@ -0,0 +1,196 @@
+                {
+                    return;
+                }
+            }
```
Unnecessary brackets here, it adds extra tabbing.

EVIJA-OZO replied:
Code part removed.

GuntarsD commented on Savanna/Model/Animal.cs:
```diff
@@ -0,0 +1,196 @@
+                else
+                {
+                    return;
+                }
```
Code:
else
{
    return;
}

is obsolete.

EVIJA-OZO replied:
Code part removed.

GuntarsD commented on Savanna/Model/Animal.cs:
```diff
@@ -0,0 +1,196 @@
+        }
+
+        /// <summary>
+        /// Returns all animals in the range of vision, excluding itself.
```
Return type is bool. So comment is wrong.

EVIJA-OZO replied:
Comment changed.

EdgarsBondars commented on Savanna/Model/Animal.cs:
```diff
@@ -0,0 +1,196 @@
+        protected static List<AnimalNewCoordinates> FreeCellsToMove(List<Animal> animals, Animal animal)
+        {
+            List<AnimalNewCoordinates> newAnimalsCoordinates = new();
+            AnimalNewCoordinates newCoordinates;
```
Declare variables where they are used (newCoordinates).

EdgarsBondars commented on Savanna/Model/Animal.cs:
```diff
@@ -0,0 +1,196 @@
+        {
+            for (int element = 0; element < animals.Count; element++)
+            {
+                Animal? animal = animals[element];
```
You can try using LINQ here to simplify like animals.Exists(....)

EVIJA-OZO replied:
LINQ used for method.

EdgarsBondars commented on Savanna/Model/Antelope.cs:
```diff
@@ -0,0 +1,127 @@
+        /// </summary>
+        public Antelope()
+        {
+            Id = GenerateId();
```
Is this Id used anywhere ?

EVIJA-OZO replied:
Thought that maybe it can be useful in next iterations.
For now Id removed.

EdgarsBondars commented on Savanna/Model/Antelope.cs:
```diff
@@ -0,0 +1,127 @@
+        /// <param name="lionsAround">List of lions in current animal vision range.</param>
+        /// <param name="antilope">Current animal.</param>
+        /// <returns></returns>
+        private Animal? FindNearestLion(List<Lion> lionsAround, Animal antelope)
```
FindNearestLion and CalculateClosestLion are similar by naming and return/input parameters. Combine these methods into one.

EVIJA-OZO replied:
Methods have been merged.

EdgarsBondars commented on Savanna/Model/Game.cs:
```diff
@@ -0,0 +1,76 @@
+        /// </summary>
+        /// <param name="animals">List of animals.</param>
+        /// <param name="args">Array containing the game field.</param>
+        public static void FillGameFieldWithAnimals(List<Animal> animals, string[,] args)
```
Rename args to to more understandable, gameField maybe.

EVIJA-OZO replied:
Renamed.

EVIJA-OZO committed:
Renamed generic names;
Removed empty and obsolete code lines;
IsCellReserved method with LINQ;
FindNearestLion and CalculateClosest lion methods merged;

EdgarsBondars commented on Savanna/Model/Animal.cs:
```diff
@@ -0,0 +1,170 @@
+        /// <returns>Is cell reserved.</returns>
+        private static bool IsCellReserved(int newRowCoordinate, int newColumnCoordinate, List<Animal> animals)
+        {
+            return animals.Where(animal => animal.RowCoordinate.Equals(newRowCoordinate) && animal.ColumnCoordinate.Equals(newColumnCoordinate)).Any();
```
This can be simplified like
```c#
return animals.Any(animal => animal.RowCoordinate.Equals(newRowCoordinate) && animal.ColumnCoordinate.Equals(newColumnCoordinate));
```
You can resolve this comment yourself when fixed.

EVIJA-OZO replied:
Ok, Thank you!

EVIJA-OZO committed:
Fixed method IsCellReserved.
